delegation/src/Pos/Delegation/Logic/Mempool.hs

```diff
@@ -173,10 +173,12 @@ processProxySKHeavyInternal psk = do
         -- by the database.
     let _cmHasPostedThisEpoch = mempty
     let cedeModifier = CedeModifier {..}
+    -- verificationError is only something sane if pskValid is false.
+    -- 'Maybe' could be used here, but it'll make if-case in the end less beautiful.
```

Comment:
I believe you refer to this line:

```
not pskValid -> PHInvalid verificationError
```

What if we keep the result as `Either` and do

```
(Left verErr <- verResult) -> PHInvalid verErr
```
This simple change will allow us to remove:

* the `error` call
* the comment that explains why this `error` call is safe
* `fmap`, `either`, `(, False)`